src/components/ui/Chat/MessageList/formatMessageList.tsx

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+
+import DateHeader from './DateHeader';
+import ChatBubble from '../ChatBubble';
+import { PopOverItemProps } from '../../PopOver/PopOverItem';
+
+export interface MessageProps {
+  /* The displayed text of the message sent. */
+  content: string;
+  /* The unique identifier of the message. */
+  messageId: string;
+  /* The timestamp when the message was originally sent by the sender. */
+  createdTimestamp: string;
+  /* The timestamp of the last time the message was edited. */
+  lastUpdatedTimestamp: string;
+  /* Determines if a message was redacted (deleted) by a user. */
+  redacted: boolean;
+  /* The unique identifier of the channel the message was sent in. */
+  channelArn?: string;
+  sender: {
+    /* The display name of the sender. */
+    name: string;
+    /* The unique identifier of the sender. */
+    arn: string;
+  }
+};
+
+export const formatMessageList = (messages: MessageProps[], memberArn: string, messageActions: PopOverItemProps[] | null) => {
```

Comment:
This function feels a bit off. The naming `formatMessageList` sounds like a utility function for formatting the message data structure, but it's doing a bit of that and additionally returning an array of ChatBubble components

IMO, this should just be a component `<FormattedMessageList/>`, or something along those lines

I would also consider stripping out these utilities and exporting them individually, so that if a user doesn't want to use this component, they can still use the logic for inserting chat bubble tails/formatting time, etc

Reply:
Dave and I talked quite a bit about how do to this, but it's still up in the air, and I was curious what you thought. I originally was going to make it a component, but I wanted something that could create styling but could also be used with the <InfiniteList/> component. With the utilities stripped out of <FormattedMessageList/> it wouldn't do anything. This was an attempt to strip these styling functions out of <InfiniteList />. I do see the value in separating them though.

Reply:
Just to be clear, by stripping out the utilities I mean make them independently usable. So `FormattedMessageList` would still be responsible for importing and applying those utility functions and rendering the component list, but the utilities themselves can be used outside of the component if a developer doesn't want to use that List component

Reply:
Yeah, I can split each function out to it's own utility, I think that makes sense. But I don't think `<FormattedMessageList />` should be a component. The component that should actually render `<li>` tags on the DOM already exists (`<InfiniteList/>`). We just need these functions to return an array. I don't think you could use a `<FormattedMessage/>` component and  an `<InfiniteList/>` component together.

Reply:
Hmmm I see. One option would be to adjust InfiniteList so it's more flexible and takes away the "list" responsibility, just manages the infinite scrolling piece of it. An InfiniteList HOC or something like that would be more flexible for future use cases, and potentially more pleasant to work with

Otherwise, we could adjust the name of the function to better indicate what it's doing. `formatMessageList` sounds like it's formatting the data, not really returning an array of components. `createChatBubbleItems` (or something similar) might help people understand the functionality a bit better